keps/sig-api-machinery/20190325-unions.md

```diff
@@ -110,15 +110,18 @@ kubebuilder types):
 - `// +union` before a structure means that the structure is a union. All the
   fields must be optional (beside the discriminator) and will be inlucded as
   members of the union. That structure CAN be embedded in another structure.
-- `// +unionDeprecated` before a field means that it is part of the
+- `// +unionMemberDeprecated` before a field means that it is part of the
   union. Multiple fields can have this prefix. These fields MUST BE optional,
   omitempty and pointers. The field is named deprecated because we don't want
   people to embed their unions directly in structures, and only exist because of
   some existing core types (e.g. `Value` and `ValueFrom` in
   [EnvVar](https://github.com/kubernetes/kubernetes/blob/3ebb8ddd8a21b/staging/src/k8s.io/api/core/v1/types.go#L1817-L1836)).
 - `// +unionDiscriminator` before a field means that this field is the
   discriminator for the union. Only one field per structure can have this
-  prefix. This field HAS TO be a string, and CAN be optional.
+  prefix. This field HAS TO be a string, and SHOULD BE required.
+- `// +deduceInvalidDiscriminatorDeprecated` is a new field that should only be
```

Comment:
This is a super confusing name. How about just "deduceDiscriminatorDeprecated"?

Reply:
In the current context, I think `Invalid` is useful ...

Reply:
I don't understand what is invalid though?

Reply:
e.g. permitting the client to ignore the discriminator is not semantically the same thing as permitting them to send invalid discriminators.

I'm also not sure why it needs to be deprecated behavior. If we don't have to force clients to understand the discriminator, why would we? It makes their life harder for what benefit?